### PR TITLE
Fix handling of sslmode=require

### PR DESCRIPTION
**Pull request by meln5674**
Remove the sslrootcert parameter from the postgres connection string if `sslmode=require`. Otherwise, this has the same effect as setting `sslmode=verify-ca`

Signed-off-by: Andrew Melnick <meln5674@kettering.edu>

See https://www.postgresql.org/docs/9.1/libpq-ssl.html

### CODE REVIEW

**Comment by robertp** [comment]
Hey @meln5674 this looks nice, we'll do some testing on it shortly. Thank you!

File: stable/anchore-engine/templates/enterprise_feeds_configmap.yaml [hunk 1]
```diff
@@ -56,12 +56,16 @@ data:
...
         db_connect: "postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}"
+        {{- else if eq .Values.anchoreEnterpriseFeeds.dbConfig.sslMode "require" }}
+        db_connect: "postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}?sslmode={{- .Values.anchoreGlobal.dbConfig.sslMode -}}"
+        {{- else }}
+        db_connect: "postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}?sslmode={{- .Values.anchoreGlobal.dbConfig.sslMode -}}&sslrootcert=/home/anchore/certs/{{- .Values.anchoreGlobal.dbConfig.sslRootCertName -}}"
         {{- end }}
-        db_connect_args:
+
+
+      db_connect_args:
```
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_configmap.yaml:68]: @meln5674 `db_connect_args:` needs to have the same indentation as `db_connect:` - this is causing the feeds service pod to fail to start.

**Review by Btodhunter — CHANGES_REQUESTED** [review]
@meln5674 Thanks for the PR! It looks good except the indentation issue I pointed out. I fixed that locally and almost everything spun up great with `sslMode=require`. 

This PR also pointed out to me that the DB config for the UI has the sslMode hard-coded. We'll need to to make some updates to support `sslMode=require`:

* `templates/enterprise_ui_deployment.yaml` - line 85 should be
```
{{ with .Values.anchoreGlobal.dbConfig.sslRootCertName }}
- name: PGSSLROOTCERT
  value: /home/anchore/certs/{{- . }}
{{- end }}
```
* `templates/enterprise_ui_secret.yaml`
```
  {{- if .Values.anchoreGlobal.dbConfig.ssl }}
  ANCHORE_APPDB_URI: 'postgresql://{{ index .Values "postgresql" "postgresUser" }}:{{ index .Values "postgresql" "postgresPassword" }}@{{ template "db-hostname" . }}/{{ index .Values "postgresql" "postgresDatabase" }}?ssl={{- .Values.anchoreGlobal.dbConfig.sslMode }}'
  {{- else }}
  ANCHORE_APPDB_URI: 'postgresql://{{ index .Values "postgresql" "postgresUser" }}:{{ index .Values "postgresql" "postgresPassword" }}@{{ template "db-hostname" . }}/{{ index .Values "postgresql" "postgresDatabase" }}'
  {{- end }}
```

File: stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml [hunk 2]
```diff
@@ -58,13 +58,17 @@ spec:
...
+            anchore-enterprise-manager db --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME} upgrade --dontask;
+        {{- else if eq .Values.anchoreGlobal.dbConfig.sslMode "require" }}
+          - |
+            anchore-enterprise-manager db --db-use-ssl --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}?sslmode={{- .Values.anchoreGlobal.dbConfig.sslMode -}} upgrade --dontask
         {{- else }}
           - |
-            anchore-enterprise-manager db --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME} upgrade --dontask;
+            anchore-enterprise-manager db --db-use-ssl --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}?sslmode={{- .Values.anchoreGlobal.dbConfig.sslMode -}}\\&sslrootcert=/home/anchore/certs/{{- .Values.anchoreGlobal.dbConfig.sslRootCertName -}} upgrade --dontask;
```
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml:69]: the white space chomping at the end of `{{- .Values.anchoreGlobal.dbConfig.sslRootCertName -}}` is causing the upgrade job to fail.

File: stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml [hunk 3]
```diff
@@ -58,13 +58,17 @@ spec:
...
         command: ["/bin/bash", "-c"]
         args:
-        {{- if .Values.anchoreGlobal.dbConfig.ssl }}
+        {{- if not .Values.anchoreGlobal.dbConfig.ssl }}
           - |
-            anchore-enterprise-manager db --db-use-ssl --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}?sslmode={{ .Values.anchoreGlobal.dbConfig.sslMode }}\\&sslrootcert=/home/anchore/certs/{{ .Values.anchoreGlobal.dbConfig.sslRootCertName }} upgrade --dontask;
+            anchore-enterprise-manager db --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME} upgrade --dontask;
+        {{- else if eq .Values.anchoreGlobal.dbConfig.sslMode "require" }}
+          - |
+            anchore-enterprise-manager db --db-use-ssl --db-connect postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/${ANCHORE_DB_NAME}?sslmode={{- .Values.anchoreGlobal.dbConfig.sslMode -}} upgrade --dontask
```
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_upgrade_job.yaml:66]: This line also has the over-zealous white space chomping on `{{- .Values.anchoreGlobal.dbConfig.sslMode -}}`

**Comment by Btodhunter** [comment]
@meln5674 There are just a few more things that need to be resolved for this to work as expected. I outlined them in the comments above this one. I made all those changes locally and everything came up perfectly with `sslMode=require`. So once those are implemented I'll merge it!

**Comment by Btodhunter** [comment]
@meln5674 I wanted to check in on this PR before it became too stale. It's a great addition to the chart, it just needs a few tweaks before we can merge. I plan on implementing these changes with the required tweaks in a couple of weeks if I don't hear back from you. Thanks!

**Pull request closed without merging**